### PR TITLE
Add recommended security headers

### LINKED ISSUE #618: Add recommended security headers
When we run [securityheaders.io for try](https://securityheaders.io/?q=https%3A%2F%2Ftry.pixelated-project.org%3A8080%2F&hide=on).

Headers that are missing:
* X-Frame-Options
* X-XSS-Protection
* X-Content-Type-Options
* Public-Key-Pins

We could add the first three without any bigger side effects, only Public-Key-Pins depends on the server setup, so we should skip it for now.

Acceptance criteria **(updated)**:
The user-agent responds with these headers and use the recommended values:
- [ ] X-Frame-Options: SAMEORIGIN
- [ ] X-XSS-Protection: 1; mode=block
- [ ] X-Content-Type-Options: nosniff

### PR DESCRIPTION
**Pull request by phss**
Fixes #618.

### CODE REVIEW

File: service/pixelated/config/site.py [hunk 1]
```diff
@@ -2,12 +2,15 @@
 
 
 class AddCSPHeaderRequest(Request):
```
Comment (fbernitt) [inline service/pixelated/config/site.py]: I think we should rename this Class, as by now it no longer only adds CSP headers. How about AddSecurityHeadersRequest?

**Comment by phss** [comment]
Thanks for the feedback. I've updated the pull request with the class rename.

However, Snap CI is failing with the following error. Is it something I've done?
`fatal: no submodule mapping found in .gitmodules for path 'service/src/leap.auth'`

**Comment by fbernitt** [comment]
Not sure what caused the build problems, we don't use submodules for the user-agent repo. Build is green now!

**Comment by fbernitt** [comment]
Thanks so much for you PR! Merged it!

**Pull request merged**